### PR TITLE
Fix #536: Allow attaching inbox message to multiple applications

### LINKED ISSUE #536: Allow attaching inbox message to multiple applications
Currently, an inbox message can be attached to only one application. This does not work since it requires calling the endpoint multiple times with knowledge of all applications.

### PR DESCRIPTION
**Pull request by petrdvorak**

### CODE REVIEW

File: powerauth-push-server/src/main/java/io/getlime/push/controller/rest/InboxController.java [hunk 1]
```diff
@@ -77,34 +77,23 @@ public ObjectResponse<GetInboxMessageCountResponse> fetchMessageCountForUser(
...
 
-    @GetMapping("users/{userId}/messages/{id}")
-    public ObjectResponse<GetInboxMessageDetailResponse> fetchMessageDetail(
+    @PutMapping("users/{userId}/read-all")
+    public Response readAllMessages(
             @NotNull @Size(min = 1, max = 255) @PathVariable("userId") String userId,
-            @NotNull @Size(min = 1, max = 255) @RequestParam("appId") String appId,
-            @NotNull @PathVariable("id") String inboxId) throws InboxMessageNotFoundException, AppNotFoundException {
-        return new ObjectResponse<>(inboxService.fetchMessageDetail(userId, appId, inboxId));
+            @NotNull @Size(min = 1, max = 255) @RequestParam("appId") String appId) throws AppNotFoundException {
```
Comment (banterCZ) [inline powerauth-push-server/src/main/java/io/getlime/push/controller/rest/InboxController.java:83]: Shouldn't we be consistent? I see `appId` here but `applications` at `users/{userId}`. What about to have `appIds` there?

File: powerauth-push-server/src/main/java/io/getlime/push/service/InboxService.java [hunk 2]
```diff
@@ -124,13 +117,18 @@ public GetInboxMessageDetailResponse readMessage(String userId, String appId, St
...
-        checkAppId(appId);
-        int countRead = inboxRepository.markAllAsRead(userId, appId);
+        final AppCredentialsEntity appCredentialsEntity = checkAppId(appId);
+        int countRead = inboxRepository.markAllAsRead(userId, appCredentialsEntity);
         logger.info("Marked all inbox messages as read for user: {}, count: {}", userId, countRead);
     }
 
-    private void checkAppId(String appId) throws AppNotFoundException {
-        appCredentialsRepository.findFirstByAppId(appId).orElseThrow(
+    private AppCredentialsEntity checkAppId(String appId) throws AppNotFoundException {
```
Comment (banterCZ) [inline powerauth-push-server/src/main/java/io/getlime/push/service/InboxService.java:125]: I would not call this method `checkAppId` anymore.

File: powerauth-push-server/src/main/java/io/getlime/push/service/apns/ApnsRejectionReason.java [hunk 3]
```diff
@@ -18,7 +18,7 @@
 
 /**
  * APNs rejection reason codes, documented on the Apple site:
- * https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/handling_notification_responses_from_apns
+ * <a href="https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/handling_notification_responses_from_apns">...</a>
```
Comment (banterCZ) [inline powerauth-push-server/src/main/java/io/getlime/push/service/apns/ApnsRejectionReason.java:21]:
```suggestion
 * <a href="https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/handling_notification_responses_from_apns">Handling Notification Responses from APNs</a>
```

**Review by banterCZ — APPROVED** [review]
Looks good.

**Review by romanstrobl — CHANGES_REQUESTED** [review]
I reviewed the changes in following PR: https://github.com/wultra/powerauth-push-server/pull/543. There are some issue we should handle before merging this PR.

File: powerauth-push-server/src/main/java/io/getlime/push/controller/rest/InboxController.java (hunk 1 above)
Comment (petrdvorak) [reply powerauth-push-server/src/main/java/io/getlime/push/controller/rest/InboxController.java:83]: In this case, it does not make much sense. Generally speaking, using just `appId` would suffice for the current states. The reason we have `applications` in the list is that for some admin interfaces (in `powerauth-cloud`), I can imagine requesting messages for multiple applications.

File: powerauth-push-server/src/main/java/io/getlime/push/service/InboxService.java (hunk 2 above)
Comment (petrdvorak) [reply powerauth-push-server/src/main/java/io/getlime/push/service/InboxService.java:125]: Agreed.

File: pom.xml [hunk 4]
```diff
@@ -76,9 +76,9 @@
         <guava.version>31.1-jre</guava.version>
         <h2.version>2.1.214</h2.version>
         <javax.interceptor-api.version>1.2.2</javax.interceptor-api.version>
-        <rest-base.version>1.6.0-SNAPSHOT</rest-base.version>
-        <powerauth.version>1.4.0-SNAPSHOT</powerauth.version>
-        <powerauth-crypto.version>1.4.0-SNAPSHOT</powerauth-crypto.version>
+        <rest-base.version>1.6.0</rest-base.version>
```
[the remark this answers is not in the document]
Comment (petrdvorak) [reply pom.xml:79]: I will resolve this conflict upon merge.

File: docs/Push-Server-Database.md [hunk 5]
```diff
@@ -262,6 +262,96 @@ CREATE INDEX push_campaign_user_detail ON push_campaign_user (user_id);
...
+| `subject`           | TEXT         | -                                 | Message subject.                                |
+| `body`              | TEXT         | -                                 | Message body.                                   |
+| `read`              | BOOLEAN      | index                             | Indication of in the message was read.          |
+| `timestamp_created` | TIMESTAMP    | -                                 | Timestamp of message creation.                  |
+| `timestamp_read`    | TIMESTAMP    | -                                 | Timestamp of when the message was read.         |
+
+#### Keys
+
+| Name | Primary | References | Description |
+|---|---|---|---|
```
[the remark this answers is not in the document]
Comment (petrdvorak) [reply docs/Push-Server-Database.md:305]: Fixed in #543.

**Review by romanstrobl — APPROVED** [review]
Approved.

**Pull request merged**